### PR TITLE
Restructuring the reactor build, Upgrading to latest Eclipse and deps

### PR DESCRIPTION
**Pull request by lhein**

### CODE REVIEW

**Comment by apupier** [comment]
CI pull request job is reporting this error:

```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-readable POM /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest/camel-sap/com.sap.conn.jco/pom.xml: /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest/camel-sap/com.sap.conn.jco/pom.xml (No such file or directory) @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project  (/home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest/camel-sap/com.sap.conn.jco/pom.xml) has 1 error
[ERROR]     Non-readable POM /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest/camel-sap/com.sap.conn.jco/pom.xml: /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest/camel-sap/com.sap.conn.jco/pom.xml (No such file or directory)
```

**Comment by lhein** [comment]
right....might be caused by the not up2date job config. with that PR only one call to mvn is needed in the main reactor.

File: camel-sap/features/org.fusesource.camel.component.sap.feature/feature.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@ permissions and limitations under the License.
    </license>
 
    <requires>
-      <import plugin="com.sap.conn.jco" version="3.0.11.redhat_1" match="greaterOrEqual"/>
+      <import plugin="com.sap.conn.jco" version="3.0.11.redhat_1" match="greaterOrEqual"/> <!--really? sounds weird imo -->
```
Comment (apupier) [inline camel-sap/features/org.fusesource.camel.component.sap.feature/feature.xml:34]: add comment to have a look before merging

**Review by apupier — DISMISSED** [review]
- What about adding the required maven repository, even if requires vpn connection?
Otherwise I got [ERROR] Failed to execute goal on project camel-sap: Could not resolve dependencies for project org.fusesource.components.camel-sap:camel-sap:bundle:6.3.0.redhat-SNAPSHOT: The following artifacts could not be resolved: com.sap.conn.jco:sapjco3:jar:3.0.11, com.sap.conn.idoc:sapidoc3:jar:3.0.10: Could not find artifact com.sap.conn.jco:sapjco3:jar:3.0.11 at specified path C:\git\fuse-components\camel-sap\plugins\camel-sap-component\target\lib\sapjco3.jar -> [Help 1]

- even with artefacts locally, if I first "mvn clean" then "mvn install"
[ERROR] Failed to execute goal on project camel-sap: Could not resolve dependencies for project org.fusesource.components.camel-sap:camel-sap:bundle:6.3.0.redhat-SNAPSHOT: The following artifacts could not be resolved: com.sap.conn.jco:sapjco3:jar:3.0.11, com.sap.conn.idoc:sapidoc3:jar:3.0.10: Could not find artifact com.sap.conn.jco:sapjco3:jar:3.0.11 at specified path C:\git\fuse-components\camel-sap\plugins\camel-sap-component\target\lib\sapjco3.jar -> [Help 1]


I think the first action is to create a job on CI to build the PR/branch to be able to start reviewing the PR in good condition.

**Comment by apupier** [comment]
I created the PR job for the new structure https://fusesource-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/fuse-components-camel-sap-6-3-x-pullrequest-new-structure

it is reproducing the same build failure that I observed locally:

```
[ERROR] Failed to execute goal on project camel-sap: Could not resolve dependencies for project org.fusesource.components.camel-sap:camel-sap:bundle:6.3.0.redhat-SNAPSHOT: The following artifacts could not be resolved: com.sap.conn.jco:sapjco3:jar:3.0.11, com.sap.conn.idoc:sapidoc3:jar:3.0.10: Could not find artifact com.sap.conn.jco:sapjco3:jar:3.0.11 at specified path /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest-new-structure/camel-sap/plugins/camel-sap-component/target/lib/sapjco3.jar -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project camel-sap: Could not resolve dependencies for project org.fusesource.components.camel-sap:camel-sap:bundle:6.3.0.redhat-SNAPSHOT: The following artifacts could not be resolved: com.sap.conn.jco:sapjco3:jar:3.0.11, com.sap.conn.idoc:sapidoc3:jar:3.0.10: Could not find artifact com.sap.conn.jco:sapjco3:jar:3.0.11 at specified path /home/****/workspace/fuse-components-camel-sap-6-3-x-pullrequest-new-structure/camel-sap/plugins/camel-sap-component/target/lib/sapjco3.jar
```

**Comment by apupier** [comment]
tests are now failing with error:

`JCo initialization failed with java.lang.ExceptionInInitializerError: Illegal JCo archive "sapjco3-3.0.11.jar". It is not allowed to rename or repackage the original archive "sapjco3.jar".`

File: camel-sap/tests/org.fusesource.camel.component.sap.test/pom.xml [hunk 2]
```diff
@@ -177,14 +177,10 @@
         <dependency>
             <groupId>com.sap.conn.jco</groupId>
             <artifactId>sapjco3</artifactId>
-            <scope>system</scope>
```
Comment (apupier) [inline camel-sap/tests/org.fusesource.camel.component.sap.test/pom.xml]: are you sure that we can remove the scope item? it means that we can't publish the org.fusesource.camel.component.sap.test plugin to a public location (but I guess it is already the case?)

File: camel-sap/features/pom.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<!--
+ Copyright 2013 Red Hat, Inc.
```
Comment (apupier) [inline camel-sap/features/pom.xml:2]: 2013 -> 2019 ?

File: camel-sap/features/org.fusesource.camel.component.sap.feature/pom.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<!--
+ Copyright 2013 Red Hat, Inc.
```
Comment (apupier) [inline camel-sap/features/org.fusesource.camel.component.sap.feature/pom.xml:2]: 2013 -> 2019?

File: camel-sap/features/org.fusesource.camel.component.sap.feature/pom.xml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	<packaging>eclipse-feature</packaging>
+
+	<name>JBoss Fuse :: Components :: SAP :: Features :: SAP Data Layer</name>
+	<description>Provides Feature for the SAP Data Layer</description>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>ianal-maven-plugin</artifactId>
```
Comment (apupier) [inline camel-sap/features/org.fusesource.camel.component.sap.feature/pom.xml:37]: is the license file a requirement for productization? If yes, maybe we can't just skip it.

File: camel-sap/pom.xml [hunk 6]
```diff
@@ -18,73 +24,367 @@
...
 	<packaging>pom</packaging>
 
-	<name>JBoss Fuse :: Components :: SAP JCO</name>
+	<name>JBoss Fuse :: Components :: SAP</name>
 	<description>Provides SAP Camel Component</description>
 	<url>http://www.jboss.org/products/fuse/overview/</url>
 
+	<organization>
+		<name>Red Hat</name>
+		<url>http://www.redhat.com</url>
```
Comment (apupier) [inline camel-sap/pom.xml:38]: http://www.redhat.com -> https://www.redhat.com

File: camel-sap/pom.xml [hunk 7]
```diff
@@ -18,73 +24,367 @@
...
+	</issueManagement>
+
+	<scm>
+		<connection>scm:git:git://git@github.com:jboss-fuse/fuse-components.git</connection>
+		<developerConnection>scm:git:ssh://git@github.com:jboss-fuse/fuse-components.git</developerConnection>
+		<url>https://github.com/jboss-fuse/fuse-components</url>
+	</scm>
+
+	<prerequisites>
+		<maven>3.6</maven>
```
Comment (apupier) [inline camel-sap/pom.xml:53]: I'm wondering if it is compatible with productization requirements for 6.3 stream. Worth checking.

File: camel-sap/pom.xml [hunk 8]
```diff
@@ -18,73 +24,367 @@
...
+
+		<maven.compiler.target>1.8</maven.compiler.target>
+		<maven.compiler.source>1.8</maven.compiler.source>
+
+		<tycho-version>1.3.0</tycho-version>
+		<maven-dependency-plugin-version>3.1.1</maven-dependency-plugin-version>
+		<maven-surefire-plugin-version>2.22.1</maven-surefire-plugin-version>
+		<maven-resources-plugin-version>3.1.0</maven-resources-plugin-version>
+
 		<camel-version>2.17.0.redhat-630133</camel-version>
```
Comment (apupier) [inline camel-sap/pom.xml:68]: this version is quite strange but not the scope of this PR (just a note for later)

File: camel-sap/pom.xml [hunk 9]
```diff
@@ -18,73 +24,367 @@
...
-			
+		<sap.idoc.lib.version>3.0.10</sap.idoc.lib.version>
+		<sap.jco.lib.version>3.0.11</sap.jco.lib.version>
+
+		<sap.redhat.version.postfix>redhat_2</sap.redhat.version.postfix>
+
+		<sap.idoc.wrapper.version>${sap.idoc.lib.version}.${sap.redhat.version.postfix}</sap.idoc.wrapper.version>
+		<sap.jco.wrapper.version>${sap.jco.lib.version}.${sap.redhat.version.postfix}</sap.jco.wrapper.version>
+
+		<jaxb.core.version>2.3.0.1</jaxb.core.version>
```
Comment (apupier) [inline camel-sap/pom.xml:79]: on 6.3 stream it was 2.2.11, I think it is something that needs to stay aligned with other Fuse dependencies

https://github.com/jboss-fuse/fuse-components/blob/94a5d126f71d7b9158210dd7c2248a345e4d428e/camel-sap/pom.xml#L181

File: camel-sap/pom.xml [hunk 10]
```diff
@@ -18,73 +24,367 @@
...
-		<!-- JBoss Tools Integration Stack version -->
-	    	<jbtis.version>4.4.3.Final</jbtis.version>
-		
-		<maven.compiler.target>1.7</maven.compiler.target>
-		<maven.compiler.source>1.7</maven.compiler.source>
-		<maven-surefire-plugin-version>2.19</maven-surefire-plugin-version>
+		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+		<maven.compiler.target>1.8</maven.compiler.target>
```
Comment (apupier) [inline camel-sap/pom.xml:60]: i think that in 6.3 stream we need to stick with 1.7 as it is a supported java version https://access.redhat.com/articles/310603#Red-Hat-JBoss-Fuse-63

File: camel-sap/pom.xml [hunk 11]
```diff
@@ -18,73 +24,367 @@
...
+		<powermock-version>1.5.6</powermock-version>
+		<hamcrest.version>1.3</hamcrest.version>
+		<hamcrest.integration.version>1.3</hamcrest.integration.version>
+
+		<eclipse.update.site.url>http://download.eclipse.org/releases/2019-03</eclipse.update.site.url>
+		<jbosstools-locus.site.url>http://repository.jboss.org/nexus/content/unzip/unzip/org/jboss/tools/locus/jbosstools-locus/1.9.0-SNAPSHOT/jbosstools-locus-1.9.0-SNAPSHOT-updatesite.zip-unzip/</jbosstools-locus.site.url>
+
+		<!-- upstream eclipse bundle versions -->
+		<emf.core.version>2.15.0</emf.core.version>
+		<emf.edit.version>2.14.0</emf.edit.version>
```
Comment (apupier) [inline camel-sap/pom.xml:90]: from 2.8.0 to 2.14.0, seems a big increase for a release patch stream https://github.com/jboss-fuse/fuse-components/blob/94a5d126f71d7b9158210dd7c2248a345e4d428e/camel-sap/pom.xml#L126

I think it is used at runtime too, not only for SAP Tooling. If only SAP tooling, i think it is fine. If at runtime, I think it is not but worth checking with Productization team.

File: camel-sap/pom.xml [hunk 12]
```diff
@@ -18,73 +24,367 @@
...
+					<groupId>org.eclipse.tycho</groupId>
+					<artifactId>tycho-surefire-plugin</artifactId>
+					<version>${tycho-version}</version>
+					<configuration>
+						<useUIThread>true</useUIThread>
+						<useUIHarness>true</useUIHarness>
+						<includes>
+							<include>**/*IT.class</include>
+						</includes>
+						<argLine>${tycho.testArgLine} -XX:+HeapDumpOnOutOfMemoryError ${platformSystemProperties} ${systemProperties} ${moduleProperties} -Dusage_reporting_enabled=false -Dorg.jboss.tools.vpe.loadxulrunner=false -Dgroovy.grape.report.downloads=true -Divy.message.logger.level=4 -Dsun.net.client.defaultConnectTimeout=30000 -Dsun.net.client.defaultReadTimeout=30000 -Djava.library.path="${native.lib.directory}"</argLine>
```
Comment (apupier) [inline camel-sap/pom.xml:280]: why do we need all these parameters?
As far as I know
- groovy.grape and ivy.message.logger.level is used by Fuse Tooling
- usage_reporting_enabled is used by Jboss Tools
- org.jboss.tools.vpe.loadxulrunner given the name it seems to be used by Jboss Tools only also

File: camel-sap/tests/pom.xml [hunk 13]
```diff
@@ -0,0 +1,36 @@
+<!--
+ Copyright 2013 Red Hat, Inc.
```
Comment (apupier) [inline camel-sap/tests/pom.xml:2]: 2013 -> 2019

**Review by apupier — CHANGES_REQUESTED** [review]
- the plugin test are not played see https://fusesource-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/fuse-components-camel-sap-6-3-x-pullrequest-new-structure/9/testReport/org.fusesource.camel.component.sap/ which is not containing TRFCTest for instance
- would worth checking with productization team how to handle the new plugins that are now part of the built artefact (have a profile to not build them? have a rule in productization to ignore them?)

**Comment by lhein** [comment]
Priority of that task has been lowered and work is paused now.

**Pull request closed without merging**